### DISCUSSION TITLE
Question: what dependencies does a compiled template function have?

### DISCUSSION THREAD

**Discussion by netaisllc** [eta-dev/eta#116] [Q&A] [answered]
### Not a bug at all

Considering using `Eta` for a project that requires templates to be precompiled into functions; Eta seems to have this well covered. Kudos on the approach and for being fresh, fast, and typed.

The ideal case is being able to take such a function and executing it in an arbitrary JS environment. But from the docs I can't really tell if the function assumes Eta, helpers, a runtime, etc to also be present. (Some other JS templating projects use that approach.)

Any hints would be helpful and I'm happy to update docs in this regard for you :). Such as here https://eta.js.org/docs/api/compilation and/or the TypeDoc page.

**Comment by shadowtime2000** [comment, marked as answer]
Eta doesn't assume an external dependencies, you could just use compile to string take the output put a name on the function and use it wherever. Well, unless your using stuff like file system access.
